app/controllers/work_packages/activities_tab_controller.rb

```diff
@@ -0,0 +1,247 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2010-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+class WorkPackages::ActivitiesTabController < ApplicationController
+  include OpTurbo::ComponentStream
+
+  before_action :find_work_package
+  before_action :find_project
+  before_action :find_journal, only: %i[edit cancel_edit update]
+  before_action :authorize
+
+  def index
+    render(
+      WorkPackages::ActivitiesTab::IndexComponent.new(
+        work_package: @work_package,
+        filter: params[:filter]&.to_sym || :all
```

Comment:
This line is copied around multiple times in the controller. Put it in a before action instead?